code/datums/martial/ultra_violence.dm

```diff
@@ -123,10 +124,15 @@
 
 ---------------------------------------------------------------*/
 /datum/martial_art/ultra_violence/proc/pocket_pistol(mob/living/carbon/human/A)
-	var/obj/item/gun/ballistic/revolver/ipcmartial/gun = new /obj/item/gun/ballistic/revolver/ipcmartial (A)   ///I don't check does the user have an item in a hand, because it is a martial art action, and to use it... you need to have a empty hand
-	gun.gun_owner = A
+	var/obj/item/gun/ballistic/revolver/ipcmartial/gun = locate() in A // check if they already had one
+	if(gun && style < 8) // can't dual wield unless you're at the max style level
+		to_chat(A, span_notice("You reload your revolver."))
+		gun.magazine.top_off()
+	else
+		gun = new(A)   ///I don't check does the user have an item in a hand, because it is a martial art action, and to use it... you need to have a empty hand
```

Comment:
won't this cause a problem if they already have a gun and have enough style?

Reply:
It gives them a new gun instead of reloading the existing one, that is intended

Reply:
Or I could make it reload the other one too if that's what you wanted me to do

Reply:
Alright, done